src/MongoDB.Driver.Core/MongoDB.Driver.Core.csproj

```diff
@@ -25,6 +25,18 @@
     <IncludeSymbols>true</IncludeSymbols>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <IsWindows Condition="$([MSBuild]::IsOSPlatform('Windows'))">true</IsWindows>
+    <IsOSX Condition="$([MSBuild]::IsOSPlatform('OSX'))">true</IsOSX>
+    <IsLinux Condition="$([MSBuild]::IsOSPlatform('Linux'))">true</IsLinux>
+  </PropertyGroup>
+
+  <Target Name="PrintOS" BeforeTargets="Build">
```

Comment:
lines 34-38 are used only for logging. In windows, you will see the used values in the build output. Can be removed without any problems :)